src/main/java/com/linecorp/armeria/common/thrift/text/TTextProtocol.java

```diff
@@ -687,6 +696,17 @@ private String getWriterString() throws TException {
         return ret;
     }
 
+    private Class<?> getCurrentFieldClassIfIs(Class<?> classToMatch) {
+        if (currentFieldClass.isEmpty() || currentFieldClass.peek() == null) {
```

Comment:
Nah peek() throws an exception when empty, not returning null. I push nulls intentionally since otherwise it wouldn't be possible to call pop() correctly. If this is confusing, I can push a marker singleton instead of nulls.